modules/porous_flow/include/utils/PorousFlowCapillaryRSC.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
+#ifndef POROUSFLOWCAPILLARYRSC_H
+#define POROUSFLOWCAPILLARYRSC_H
+
+#include "GeneralUserObject.h"
+
+/**
+ * Rogers-Stallybrass-Clements version of effective saturation as a function of CAPILLARY pressure.
+ * valid for residual saturations = 0, and viscosityOil = 2*viscosityWater.  (the "2" is important here!).
+ * C Rogers, MP Stallybrass and DL Clements "On two phase filtration under gravity and with boundary infiltration: application of a Backlund transformation" Nonlinear Analysis Theory Methods and Applications 7 (1983) 785--799.
+ */
+class PorousFlowCapillaryRSC
+{
+public:
+
+  /**
+   * effective saturation as a function of capillary pressure
+   * @param pc capillary pressure
+   * @param shift RSC's shift parameter
+   * @param scale RSC's scale parameter
+   */
+  static Real seff(Real pc, Real shift, Real scale);
+
+  /**
+   * derivative of effective saturation wrt capillary pressure
+   * @param pc capillary pressure
+   * @param shift RSC's shift parameter
+   * @param scale RSC's scale parameter
+   */
+  static Real dseff(Real pc, Real shift, Real scale);
+
+  /**
+   * 2nd derivative of effective saturation wrt capillary pressure
+   * @param pc capillary pressure
+   * @param shift RSC's shift parameter
+   * @param scale RSC's scale parameter
+   */
+  static Real d2seff(Real pc, Real shift, Real scale);
+
+};
+
+#endif // POROUSFLOWCAPILLARYRSC_H
```

Comment:
Same deal here. I'm not sure I understand the design intentions here. Another option would be to make this a mixin class :-/